File: backend/app/elements/gui/widgets/divider/divider_element.py
"""The `divider` widget: a rule between sections. Holds nothing at all."""

from __future__ import annotations

from app.elements.base import StaticWidget
from app.models.graph import GuiWidgetKind


class DividerElement(StaticWidget):
    widget_kind = GuiWidgetKind.DIVIDER

File: frontend/src/elements/gui/widgets/divider/dividerElement.ts
import type { GuiWidgetElementDefinition } from '../../../types';
import DividerWidget from '../../../../components/gui/widgets/DividerWidget';

/** A rule between sections. No ports, no content. */
export const dividerElement: GuiWidgetElementDefinition = {
  widgetKind: 'divider',
  RuntimeWidget: DividerWidget,
};
